src/Forms/UrlField.php

```diff
@@ -7,26 +7,89 @@
 
 /**
  * Text input field with validation for a url
- * Url must include a scheme, either http:// or https://
+ * Url must include a protocol (aka scheme) such as https:// or http://
  */
 class UrlField extends TextField
 {
+    /**
+     * The default set of protocols allowed for valid URLs
+     */
+    private static array $default_protocols = ['https', 'http'];
+
+    /**
+     * The default value for whether a relative protocol (// on its own) is allowed
+     */
+    private static bool $default_allow_relative_protocol = false;
+
+    private array $protocols = [];
+
+    private ?bool $allowRelativeProtocol = null;
+
     public function Type()
     {
         return 'text url';
     }
 
     public function validate($validator)
     {
-        $result = true;
-        if ($this->value && !ConstraintValidator::validate($this->value, new Url())->isValid()) {
-            $validator->validationError(
-                $this->name,
-                _t(__CLASS__ . '.INVALID', 'Please enter a valid URL'),
-                'validation'
-            );
-            $result = false;
+        $allowedProtocols = $this->getAllowedProtocols();
+        $message = _t(
+            __CLASS__ . '.INVALID_WITH_PROTOCOL',
+            'Please enter a valid URL including a protocol, e.g {protocol}://example.com',
+            ['protocol' => $allowedProtocols[0]]
+        );
```

Comment:
New message which gives a clearer indication of what may be wrong with the URL. Most likely missing a protocol will be the common mistake.